### PR TITLE
add factory for dropin files

### PR DESCRIPTION
**Pull request by TheMeier**
Add a factory for dropin files so they can be simply defined in hiera

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -108,6 +110,7 @@
   Systemd::JournaldSettings                              $journald_settings,
   Boolean                                                $manage_logind,
   Systemd::LogindSettings                                $logind_settings,
+  Optional[Hash]                                         $dropin_files = undef,
```
Comment (raphink) [inline manifests/init.pp:113]: It'd be more simple to have:

```suggestion
  Hash                                                   $dropin_files = {},
```

File: manifests/init.pp [hunk 2]
```diff
@@ -137,4 +140,12 @@
...
     contain systemd::logind
   }
+
+  if $dropin_files {
+    $dropin_files.each |$name, $resource| {
+      systemd::dropin_file { $name:
+        * => $resource
+      }
+    }
+  }
```
Comment (raphink) [inline manifests/init.pp:150]: If the Hash is not optional but defaults to `{}`, then you don't need the condition anymore.
```suggestion

  $dropin_files.each |$name, $resource| {
    systemd::dropin_file { $name:
      * => $resource,
    }
  }
```

Ideally even, it'd be good to have a data type for that, like it was done for `$journald_settings` and `$logind_settings` above.

**Comment by TheMeier** [comment]
I have made the changes with the empty hash default. About dedicated type, isn't that a repitition as the structure of the variables is already set in the definition of the defined resource.

File: manifests/init.pp [hunk 3]
```diff
@@ -137,4 +140,10 @@
   if $manage_logind {
     contain systemd::logind
   }
+
+  $dropin_files.each |$name, $resource| {
+    systemd::dropin_file { $name:
+      * => $resource
```
Comment (raphink) [inline manifests/init.pp:146]:
```suggestion
      * => $resource,
```

**Pull request merged**